### DISCUSSION TITLE
Need Help to Creat Pull Request for Skyculture

### DISCUSSION THREAD

**Discussion by santoshasalunke** [Stellarium/stellarium#3016] [General]
Hi

I had created a new skyculture and want to include it in Stellarium on all OS platforms.
To make it part of Stellarium, I have been told to reate 'Pull Request'. I tried but it failed, that PR is closed.

As this is my first time working with GITHUB, i spent a week to read and understand this PR thing. I am not successfull.

Therefore i need somones help to add my sky culture in Stellarium.

Thank you.

**Comment by gzotti** [comment]
Have you seen https://github.com/Stellarium/stellarium/wiki/Git-Contributor-Workflow? 

`git clone https://github.com/Stellarium/stellarium.git`

(cloning to local directory)

Follow the other steps in "Preparing your fork". 

Then do the actual work on your side:

`cd stellarium`

Create a topic branch. Its name is a string that in this case looks like a directory:

`git checkout -b skycultures/myNewSkyculture`

 Now you are on your own branch and are free to make changes, e.g. add a new skyculture. Use a better fitting name than this example!

`mkdir skycultures/mySkyculture`

Now fill this directory with the actual files of your new skyculture. 
Copy and add a CMakeLists.txt file from a neighbouring skyculture directory and edit to adapt it. 
Then add these files to git:

```
git add skycultures/mySkyculture/info.ini
git add skycultures/mySkyculture/CMakeLists.txt
git add skycultures/mySkyculture/description.en.utf8
git add skycultures/mySkyculture/constellationship.fab
git add skycultures/mySkyculture/star_names.fab
git add skycultures/mySkyculture/... (whatever other files you have)
```

Then commit the new changes locally.

`git commit -m "Added new skyculture 'mySkyculture'"`

This stores the data in your `.git` repository on your own computer. 
This is the same as is described in Point 2 of the FAQ linked above. Now follow Step 3 of these instructions.

At this point I am not sure what happens, as I don't have a private fork. Maybe @10110111 can continue/add details?

**Comment by 10110111** [comment]
> `git clone https://github.com/Stellarium/stellarium.git`

This is not a great idea, since it will only get a read-only copy. You should

1. Fork the Stellarium repository (via the GitHub website);
2. [Set up your SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys#checking-for-existing-ssh-keys);
3. And then clone your fork via SSH:
```
git clone git@github.com:santoshasalunke/stellarium.git
```
After this you can continue as @gzotti described, but omitting the unnecessary `git checkout -b` branch creation step. After committing you can push your local changes to the GitHub servers:
```
git push
```
And then you can go to the GitHub page of your fork, and click _Contribute_ → _Open pull request_.

Feel free to ask if you encounter problems, I'll try to assist.

**Reply by gzotti** [reply]
Thank you @10110111. Can you please update the FAQ entry with the currently best workflow? Git is not an easy tool, and some functions are used rarely enough to require some working guidelines.  I still think contributors should make a feature branch and not change their master. There should probably also be a description of updating/rebasing the private clone with our own master.

**Reply by 10110111** [reply]
Regarding changing the master, it's not as bad for a newbie as adding yet another step of creating a branch, and then one more complication to push this into a new remote branch, then making sure to create the pull request from this specific branch instead of master, etc. Just use master, get used to the basic workflow, and then you can progress further to more advanced usage, if you really intend to continue to develop.

**Reply by 10110111** [reply]
The FAQ is OK for the developers that are going to contribute more than one change. I just fixed the broken links to GitHub help.

**Comment by santoshasalunke** [comment]
Thanks to @gzotti and @10110111 
Followed your instructions and created pull request = Added new skyculture 'marathi' #3053
